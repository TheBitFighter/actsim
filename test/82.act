import globals;
import std;
open std::channel;

defproc src (ts_bd!<1> x)
{
  chp {
    *[ x!1; x!0 ]
  }
}

defproc buffer(ts_bd?<1> l; ts_bd!<1> r)
{
  int b;
  int k;
  chp {
     k := 0;
    *[ k < 100 -> l?b; log("got ", b); r!b; k := k + 1 ]
  }
}

defproc sink (ts_bd?<1> x)
{
  prs {
  ~Reset & x.r -> x.a+
   Reset | ~x.r -> x.a-
  }
}

defproc test ()
{
   src s;
   buffer b(s.x);
   sink t(b.r);
}

Initialize {
  actions { Reset+ };
  actions { Reset- }
}
