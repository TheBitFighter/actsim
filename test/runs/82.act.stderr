WARNING: buffer<>: substituting chp model (requested prs, not found)
WARNING: src<>: substituting chp model (requested prs, not found)
